In [1]:
from train_classifieur import train_classifier, pred_classifier

Au vue de la puissance de nos ordinateurs (1 epoch prend 20 minutes), nous avons décidé de limitter le classifier à 2 epochs.

Training avec les metadata originelles avec les weights entièrement à 1 :

In [2]:
ckpt_path, ckpt_score = train_classifier(
    logdir="./expe_log/",
    datadir="./data/",
    csv="./data/metadata.csv",
    max_epochs = 2
)

/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/lightning_fabric/plugins/environments/slurm.py:204: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM, prepend your python command with `srun` like so: srun python /home/lenora/Fairness/myenvfair/lib/python3.9/site-p ...
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs


Start training


/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:654: Checkpoint directory /home/lenora/Fairness/Fairness/expe_log exists and is not empty.
/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/torch/optim/lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(

  | Name  | Type                  | Params | Mode 
--------------------------------------------------------
0 | model | ResNet                | 11.2 M | train
1 | bcm   | BinaryConfusionMatrix | 0      | train
--------------------------------------------------------
11.2 M    Trainable params
0         Non-trainable params
11.2 M    Total params
44.710    Total estimated model params size (MB)
69        Modules in train mode
0         Modules in eval mode


Sanity Checking: |          | 0/? [00:00<?, ?it/s]

/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:425: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.


/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.


Epoch 1: 100%|██████████| 128/128 [19:17<00:00,  0.11it/s, v_num=3]

`Trainer.fit` stopped: `max_epochs=2` reached.


Epoch 1: 100%|██████████| 128/128 [19:19<00:00,  0.11it/s, v_num=3]
End of training 2726.562115430832


In [4]:
pred_classifier(
    datadir="./data/",
    csv_in="./data/metadata.csv",
    csv_out="./expe_log/preds.csv",
    ckpt_path = ckpt_path
)

Start prediction on train dataset
Predictions done in 347.4760377407074
Start prediction on validation dataset
Predictions done in 447.2031879425049
0.7010193986838144
0.706847224871257


Les prédictions de l'ensemble train ont un score de 0.701


Les prédictions de l'ensemble validation ont un score de 0.706

Prédictions sur les données reweighted avec 3 attributs sensibles (Age, Follow-up et View position)

In [3]:
ckpt_path, ckpt_score = train_classifier(
    logdir="./expe_log/",
    datadir="./data/",
    csv="./data/metadata_rw1.csv",
    max_epochs = 2
)

/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/lightning_fabric/plugins/environments/slurm.py:204: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM, prepend your python command with `srun` like so: srun python /home/lenora/Fairness/myenvfair/lib/python3.9/site-p ...
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs


Start training


/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:654: Checkpoint directory /home/lenora/Fairness/Fairness/expe_log exists and is not empty.
/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/torch/optim/lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(

  | Name  | Type                  | Params | Mode 
--------------------------------------------------------
0 | model | ResNet                | 11.2 M | train
1 | bcm   | BinaryConfusionMatrix | 0      | train
--------------------------------------------------------
11.2 M    Trainable params
0         Non-trainable params
11.2 M    Total params
44.710    Total estimated model params size (MB)
69        Modules in train mode
0         Modules in eval mode


Sanity Checking: |          | 0/? [00:00<?, ?it/s]

/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:425: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.


/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.


Epoch 1: 100%|██████████| 128/128 [29:01<00:00,  0.07it/s, v_num=4]

`Trainer.fit` stopped: `max_epochs=2` reached.


Epoch 1: 100%|██████████| 128/128 [29:02<00:00,  0.07it/s, v_num=4]
End of training 3528.510983943939


In [4]:
pred_classifier(
    datadir="./data/",
    csv_in="./data/metadata_rw1.csv",
    csv_out="./expe_log/preds_rw1.csv",
    ckpt_path = ckpt_path
)

Start prediction on train dataset
Predictions done in 586.4826426506042
Start prediction on validation dataset
Predictions done in 767.4870607852936
0.6387243486150966
0.6198741178714476


Les prédictions de l'ensemble train ont un score de 0.639


Les prédictions de l'ensemble validation ont un score de 0.620

Prédictions sur les données reweighted avec comme attribut sensible Age

In [7]:
ckpt_path, ckpt_score = train_classifier(
    logdir="./expe_log/",
    datadir="./data/",
    csv="./data/weights_age.csv",
    max_epochs = 2
)

/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/lightning_fabric/plugins/environments/slurm.py:204: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM, prepend your python command with `srun` like so: srun python /home/lenora/Fairness/myenvfair/lib/python3.9/site-p ...
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs


/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:654: Checkpoint directory /home/lenora/Fairness/Fairness/expe_log exists and is not empty.
/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/torch/optim/lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(

  | Name  | Type                  | Params | Mode 
--------------------------------------------------------
0 | model | ResNet                | 11.2 M | train
1 | bcm   | BinaryConfusionMatrix | 0      | train
--------------------------------------------------------
11.2 M    Trainable params
0         Non-trainable params
11.2 M    Total params
44.710    Total estimated model params size (MB)
69        Modules in train mode
0         Modules in eval mode


Start training
Sanity Checking: |          | 0/? [00:00<?, ?it/s]

/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:425: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.


/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.


Epoch 1: 100%|██████████| 128/128 [16:44<00:00,  0.13it/s, v_num=5]

`Trainer.fit` stopped: `max_epochs=2` reached.


Epoch 1: 100%|██████████| 128/128 [16:45<00:00,  0.13it/s, v_num=5]
End of training 2514.666804790497


In [8]:
pred_classifier(
    datadir="./data/",
    csv_in="./data/weights_age.csv",
    csv_out="./expe_log/preds_age.csv",
    ckpt_path = ckpt_path
)

Start prediction on train dataset
Predictions done in 293.53436064720154
Start prediction on validation dataset
Predictions done in 378.5651767253876
0.7095737860717
0.7140949837879077


Les prédictions de l'ensemble train ont un score de 0.710


Les prédictions de l'ensemble validation ont un score de 0.714

Prédictions sur les données reweighted avec comme attribut sensible Follow-up

In [10]:
ckpt_path, ckpt_score = train_classifier(
    logdir="./expe_log/",
    datadir="./data/",
    csv="./data/weights_followup.csv",
    max_epochs = 2
)

/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/lightning_fabric/plugins/environments/slurm.py:204: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM, prepend your python command with `srun` like so: srun python /home/lenora/Fairness/myenvfair/lib/python3.9/site-p ...
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:654: Checkpoint directory /home/lenora/Fairness/Fairness/expe_log exists and is not empty.
/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/torch/optim/lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(

  | Name  | Type                  | Params | Mode 
--------------------------------------------------------
0 | model | ResNet  

Start training
Sanity Checking: |          | 0/? [00:00<?, ?it/s]

/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:425: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.


/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.


Epoch 1: 100%|██████████| 128/128 [25:36<00:00,  0.08it/s, v_num=6]

`Trainer.fit` stopped: `max_epochs=2` reached.


Epoch 1: 100%|██████████| 128/128 [25:37<00:00,  0.08it/s, v_num=6]
End of training 2646.7283387184143


In [11]:
pred_classifier(
    datadir="./data/",
    csv_in="./data/weights_followup.csv",
    csv_out="./expe_log/preds_followup.csv",
    ckpt_path = ckpt_path
)

Start prediction on train dataset
Predictions done in 457.6915850639343
Start prediction on validation dataset
Predictions done in 539.381680727005
0.6883695970538415
0.6950219340072478


Les prédictions de l'ensemble train ont un score de 0.688


Les prédictions de l'ensemble validation ont un score de 0.695

Prédictions sur les données reweighted avec comme attribut sensible View Position

In [2]:
ckpt_path, ckpt_score = train_classifier(
    logdir="./expe_log/",
    datadir="./data/",
    csv="./data/weights_view.csv",
    max_epochs = 2
)

/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/lightning_fabric/plugins/environments/slurm.py:204: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM, prepend your python command with `srun` like so: srun python /home/lenora/Fairness/myenvfair/lib/python3.9/site-p ...
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs


/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:654: Checkpoint directory /home/lenora/Fairness/Fairness/expe_log exists and is not empty.
/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/torch/optim/lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(

  | Name  | Type                  | Params | Mode 
--------------------------------------------------------
0 | model | ResNet                | 11.2 M | train
1 | bcm   | BinaryConfusionMatrix | 0      | train
--------------------------------------------------------
11.2 M    Trainable params
0         Non-trainable params
11.2 M    Total params
44.710    Total estimated model params size (MB)
69        Modules in train mode
0         Modules in eval mode


Start training
Sanity Checking: |          | 0/? [00:00<?, ?it/s]

/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:425: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.


/home/lenora/Fairness/myenvfair/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.


Epoch 1: 100%|██████████| 128/128 [23:09<00:00,  0.09it/s, v_num=8]

`Trainer.fit` stopped: `max_epochs=2` reached.


Epoch 1: 100%|██████████| 128/128 [23:09<00:00,  0.09it/s, v_num=8]
End of training 2421.5503668785095


In [3]:
pred_classifier(
    datadir="./data/",
    csv_in="./data/weights_view.csv",
    csv_out="./expe_log/preds_view.csv",
    ckpt_path = ckpt_path
)

Start prediction on train dataset
Predictions done in 353.1675353050232
Start prediction on validation dataset
Predictions done in 458.46593403816223
0.6691117283078227
0.6803356856761397


Les prédictions de l'ensemble train ont un score de 0.669


Les prédictions de l'ensemble validation ont un score de 0.680

Le reweighting apportant le meilleur résultats de prédictions est vle reweighting avec l'âge